src/main/pages/che-7/end-user-guide/proc_writing-a-devfile-for-your-project.adoc

```diff
@@ -1,27 +1,81 @@
 [id="writing-a-devfile-for-your-project_{context}"]
 = Writing a Devfile for your project
 
-This paragraph is the procedure module introduction: a short description of the procedure.
+This section describes how to create a minimal devfile for your project and how to include more than one projects in a devfile.
 
-[discrete]
-.Prerequisites
+// .Prerequisites
+// 
+// * A bulleted list of conditions that must be satisfied before the user starts following this assembly.
+// * You can also link to other modules or assemblies the user must follow before starting this assembly.
+// * Delete the section title and bullets if the assembly has no prerequisites.
 
-* A bulleted list of conditions that must be satisfied before the user starts following this assembly.
-* You can also link to other modules or assemblies the user must follow before starting this assembly.
-* Delete the section title and bullets if the assembly has no prerequisites.
 
-[discrete]
-.Procedure
+[id="preparing-a-minimal-devfile_{context}"]
+== Preparing a minimal devfile
 
-. Start each step with an active verb.
+A minimal devfile sufficient to run a workspace consists of the following parts:
 
-. Include one command or action per step.
+* Specification version
+* Name
+
+Without any further configuration, a workspace with the default editor will be launched along with its default plug-ins, which are configured on Che Server. Che-Theia is configured as the default editor along with the *Che Machine Exec* plug-in.
+
+To get a more functional workspace, the following parts can be added:
+
+* List of components: development components and user runtimes
+* List of projects: source code repositories
+* List of commands: actions to manage the workspace components, such as running the development tools, starting the runtime environments, and others
+
+.Example of a minimal devfile with a project and a standard set of plug-ins (Chhe-Theia editor, exec plug-in)
+[source,yaml]
+----
+apiVersion: 1.0.0
+metadata:
+  name: petclinic-dev-environment
+projects:
+  - name: petclinic
+    source:
+      type: git
+      location: 'https://github.com/che-samples/web-java-spring-petclinic.git'
+components:
```

Comment:
components part can be omitted in minimal devfile.

Reply:
+1 to @skabashnyuk comment (no need to specify theia and machine-exec because they are automatically added) but you may want to add java support:

```yaml
components:
  -
    type: chePlugin
    id: redhat/java/latest
```

Reply:
And it may make more sense to reference a well know example (not the one we own) like the original spring petclinic:

```yaml
projects:
  - name: petclinic
    source:
      type: git
      location: 'https://github.com/spring-projects/spring-petclinic.git'
```

Reply:
ok